client.go

```diff
@@ -16,185 +17,184 @@ import (
+//
+// The following example shows how to evaluate flags for an identity with some application-defined traits:
+//
+//	evalCtx := flagsmith.NewEvaluationContext(
+//		"my-user-123",
+//		map[string]interface{}{
+//			"company": "ACME Corporation",
+//		},
+//	)
+//	userFlags, _ := GetFlags(ctx.Background(), evalCtx)
+//	if userFlags.IsFeatureEnabled("my_feature_key") {
+//		// Flag is enabled for this evaluation context
+//	}
+func (c *Client) GetFlags(ctx context.Context, ec EvaluationContext) (f Flags, err error) {
+	if ec.identifier == "" {
+		f, err = c.GetEnvironmentFlags(ctx)
 	} else {
-		if f, err = c.GetEnvironmentFlagsFromAPI(ctx); err == nil {
-			return f, nil
-		}
+		f, err = c.GetIdentityFlags(ctx, ec.identifier, ec.traits)
 	}
-	if c.offlineHandler != nil {
-		return c.getEnvironmentFlagsFromEnvironment()
+	if err == nil {
+		return f, nil
 	} else if c.defaultFlagHandler != nil {
 		return Flags{defaultFlagHandler: c.defaultFlagHandler}, nil
+	} else {
+		return Flags{}, errors.New("GetFlags failed and no default flag handler was provided")
 	}
-	return Flags{}, &FlagsmithClientError{msg: fmt.Sprintf("Failed to fetch flags with error: %s", err)}
 }
 
-// Returns `Flags` struct holding all the flags for the current environment for
-// a given identity.
-//
-// If local evaluation is disabled it will also upsert all traits to the
-// Flagsmith API for future evaluations. Providing a trait with a value of nil
-// will remove the trait from the identity if it exists.
-//
-// If local evaluation is enabled this function will not call the Flagsmith API
-// directly, but instead read the asynchronously updated local environment or
-// use the default flag handler in case it has not yet been updated.
-//
-// Deprecated: Use `GetFlags` providing `EvaluationContext.Identity` instead.
-func (c *Client) GetIdentityFlags(ctx context.Context, identifier string, traits []*Trait) (f Flags, err error) {
-	if c.config.localEvaluation || c.config.offlineMode {
-		if f, err = c.getIdentityFlagsFromEnvironment(identifier, traits); err == nil {
-			return f, nil
-		}
-	} else {
-		if f, err = c.GetIdentityFlagsFromAPI(ctx, identifier, traits); err == nil {
-			return f, nil
-		}
+// UpdateEnvironment fetches the current environment state from the Flagsmith API. It is called periodically when using
+// [WithLocalEvaluation], or when [WithRealtime] is enabled and an update event was received.
+func (c *Client) UpdateEnvironment(ctx context.Context) error {
+	var env environments.EnvironmentModel
+	resp, err := c.client.
+		NewRequest().
+		SetContext(ctx).
+		SetResult(&env).
+		ForceContentType("application/json").
+		Get(c.baseURL + "environment-document/")
+
+	if err != nil {
+		return c.handleError(&APIError{Err: err})
 	}
-	if c.offlineHandler != nil {
-		return c.getIdentityFlagsFromEnvironment(identifier, traits)
-	} else if c.defaultFlagHandler != nil {
-		return Flags{defaultFlagHandler: c.defaultFlagHandler}, nil
+	if resp.IsError() {
```

Comment:
Same as the other comment... what happens in case of 3xx?

Reply:
Good catch also, fixed 👍